### PR TITLE
refactor: Refactor Editor into TypeScript with Smaller Components and Separate Concerns

### LINKED ISSUE #810: Refactor Editor to ts and relevant components
Refactoring editor.jsx is crucial. 
Currently, the component is a monolithic piece of code, containing 6-7 useEffect hooks, more than 12 states/setStates, and convoluted logic involving refs. It lacks clear separation between the Monaco components, completely unrelated helper functions such as debounce(), and others such as attachTypesToMonaco. It even attaches various tasks like file switching and local storage management in the same file. It currently handles both the editor functionalities, indexer actions (such as publish, register, fork, and delete), alongside code reformatting, code generation, and error handling, all within a single component.
There is odd behavior, including multiple renders on load and repeated code execution during swaps, due to the excessive stateful logic.

**GOAL**

**Separate Concerns**
Break down the component into smaller, focused components.
Isolate Monaco components, helper functions, and various tasks.
**Convert to TypeScript**
Convert the code from JSX to TSX.
Ensure type safety throughout the component.
**Class-based Components**
Implement class-based components for local storage-related functionalities.
Encapsulate state and behavior effectively.
**Optimize Stateful Logic**
Reduce the number of states/setStates.
Simplify useEffect hooks and ref usage.
Eliminate unnecessary re-renders.
**Improve Performance**
Minimize repeated code execution during swaps.
Ensure smooth and efficient component rendering.
 **Maintainence**
Ensure clear hierarchy and structure in the refactored code.
Make the codebase easier to understand, test, and extend.

### PR DESCRIPTION
**Pull request by Kevin101Zhang**
Refactored Editor Component to TypeScript. This refactoring involved breaking down the Editor file into smaller chunks and separating concerns into distinct components. Also did some minor work on validator to ts as it is a major consumer in the editor. It is setup to later iterate on some additional test for validators.

### CODE REVIEW

**Comment by Kevin101Zhang** [comment]
Still a lot of work if we want the entire Editor to be refactored. I simply just grabbed a few large components such as a few modals and localStorage and removed them to their own separate files ontop of providing type safety. This current interation with TS allows me to more easily integrate new features.

File: frontend/src/components/Editor/Legacy/primitives.d.ts [hunk 1]
```diff
@@ -0,0 +1,908 @@
+import type * as borsh from 'borsh';
```
Comment (morgsmccauley) [inline frontend/src/components/Editor/Legacy/primitives.d.ts:1]: Should this file be in `Legacy/`? We need it right?

File: frontend/src/components/Modals/ModalsView/PublishFormView.jsx [hunk 2]
```diff
@@ -96,7 +96,7 @@ const PublishFormView = ({
         id="contractFilter"
         type="text"
         placeholder="social.near"
-        value={startBlock === 'startBlockContinue' ? indexerDetails.rule.affected_account_id : contractFilter}
+        value={startBlock === 'startBlockContinue' ? contractFilter : indexerDetails.rule.affected_account_id}
```
Comment (morgsmccauley) [inline frontend/src/components/Modals/ModalsView/PublishFormView.jsx:99]: Why did this change?

File: frontend/src/components/CreateNewIndexer/CreateNewIndexer.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import Editor from '@/components/Editor';
+import Editor from '@/components/Editor/Legacy/Editor';
```
Comment (morgsmccauley) [inline frontend/src/components/CreateNewIndexer/CreateNewIndexer.js:1]: What is the purpose of Legacy?

Comment (Kevin101Zhang) [reply frontend/src/components/CreateNewIndexer/CreateNewIndexer.js:1]: Hey Morgan its just a container for what ill continue to refactor. The Legacy folder is a bit misleading in name but its simply a container for all the files I have yet to convert to TS or have not organized that are related to Editor.

File: frontend/src/components/Editor/Legacy/primitives.d.ts (hunk 1 above)
Comment (Kevin101Zhang) [reply frontend/src/components/Editor/Legacy/primitives.d.ts:1]: Yes we use everything is Legacy its nothing more than a placeholder. Maybe I should rename it to something else such as EditorComponents and continue to refactor them out as I go?

File: frontend/src/components/Modals/ModalsView/PublishFormView.jsx (hunk 2 above)
Comment (Kevin101Zhang) [reply frontend/src/components/Modals/ModalsView/PublishFormView.jsx]: Reverted back to [Contract Filter is now Editable](https://github.com/near/queryapi/pull/824/files). I must have selected the wrong option in the merge.

Comment (Kevin101Zhang) [reply frontend/src/components/Modals/ModalsView/PublishFormView.jsx]: Thanks for catching this!

File: frontend/src/components/Editor/Legacy/primitives.d.ts (hunk 1 above)
Comment (morgsmccauley) [reply frontend/src/components/Editor/Legacy/primitives.d.ts:1]: Legacy makes me assume it is being phased out - and we still need this file. We could either move this one or rename Legacy?

File: frontend/src/components/Modals/ModalsView/PublishFormView.jsx (hunk 2 above)
Comment (morgsmccauley) [reply frontend/src/components/Modals/ModalsView/PublishFormView.jsx]: I'm slightly concerned there may be other cases of incorrect merge resolves, could you please do a quick scan to check? 🙏🏼

**Review by Kevin101Zhang — COMMENTED** [review]
reviewed files

File: frontend/src/components/Modals/ModalsView/PublishFormView.jsx (hunk 2 above)
Comment (Kevin101Zhang) [reply frontend/src/components/Modals/ModalsView/PublishFormView.jsx]: I reviewed the files and the PRs between the last few pushes to dev to make sure they are present. The changes to the **contract filter** and the changes to **polling on latest indexer block** are present ontop of prettier. I also went ahead and reviewed the files and there are no noteable differences.

**Review by morgsmccauley — APPROVED** [review]
LGTM, but can you update the PR title before merging, it should use `refactor: ` not `feat: `

**Pull request merged**